lib/plugin.js

```diff
@@ -5,14 +5,23 @@ import experiments from '<%= options.experimentsDir %>'
 const MAX_AGE = <%= options.maxAge %>
 
 export default function (ctx, inject) {
+  ctx.app.router.beforeEach((to, from, next) => {
```

Comment:
Won't this override any existing function on beforeEach? Just wondering.

Reply:
`beforeEach` is a register that adds callback function, so it won't override existing functions, i think.

https://github.com/vuejs/vue-router/blob/v3.0.2/src/index.js#L121-L123